src/Microsoft.VisualStudio.SlnGen/SlnFile.cs

```diff
@@ -441,9 +443,21 @@ internal void Save(string rootPath, TextWriter writer, bool useFolders, bool col
 
             if (hierarchy != null)
             {
+                HashSet<string> drives = new ();
                 foreach (SlnFolder folder in hierarchy.Folders)
                 {
-                    string projectSolutionPath = (useFolders ? folder.FullPath.ToRelativePath(rootPath) : folder.FullPath).ToSolutionPath();
+                    bool hasPathRoot = false;
+                    if (!string.IsNullOrEmpty(folder.FullPath))
+                    {
+                        string pathRoot = Path.GetPathRoot(folder.FullPath);
+                        hasPathRoot = !string.IsNullOrEmpty(pathRoot);
+                        if (hasPathRoot)
+                        {
+                            drives.Add(pathRoot);
+                        }
+                    }
+
+                    string projectSolutionPath = (useFolders && hasPathRoot ? folder.FullPath.ToRelativePath(rootPath) : folder.FullPath).ToSolutionPath();
```

Comment:
Does this end up using absolute paths if the projects are on different drives?

Reply:
This uses absolute paths for projects that have a null or empty `FullPath`, which does occur if projects are on different drives.